datastore/longhorn.go

```diff
@@ -348,36 +348,49 @@ func (s *DataStore) ValidateSetting(name, value string) (err error) {
 		if !volumesDetached {
 			return &types.ErrorInvalidState{Reason: "cannot modify priority class setting before all volumes are detached"}
 		}
-	case types.SettingNameGuaranteedInstanceManagerCPU:
-		guaranteedInstanceManagerCPU, err := s.GetSettingWithAutoFillingRO(types.SettingNameGuaranteedInstanceManagerCPU)
+	case types.SettingNameGuaranteedInstanceManagerCPU, types.SettingNameV2DataEngineGuaranteedInstanceManagerCPU:
+		guaranteedInstanceManagerCPU, err := s.GetSettingWithAutoFillingRO(sName)
 		if err != nil {
 			return err
 		}
-		if sName == types.SettingNameGuaranteedInstanceManagerCPU {
-			guaranteedInstanceManagerCPU.Value = value
-		}
-		if err := types.ValidateCPUReservationValues(guaranteedInstanceManagerCPU.Value); err != nil {
+		guaranteedInstanceManagerCPU.Value = value
+		if err := types.ValidateCPUReservationValues(sName, guaranteedInstanceManagerCPU.Value); err != nil {
 			return err
 		}
 	case types.SettingNameStorageNetwork:
-		volumesDetached, err := s.AreAllVolumesDetached()
+		volumesDetached, err := s.AreAllVolumesDetached("")
```

Comment:
nit: Using `"""` not quite intuitive. I would suggest having a specific type for that, since unfortunately, golang doesn't support default value ..

```go
	BackendStoreDriverTypeAll = BackendStoreDriverType("all")
```

Or even have a dedicated method for it `AreAllVolumesDetached` and `AreAllVolumesDetachedByDataEngine`. BTW, as discussed before, we should make a consistent name here `DataEngine` or `BackendStore`. IIRC, you decided to use `DataEngine`

Reply:
> BTW, as discussed before, we should make a consistent name here DataEngine or BackendStore. IIRC, you decided to use DataEngine

Rename BackendStoreDriver to DataEngine seems not quite easy... I prabaly continue using BackendStoreDriver

Reply:
That's OK. However, in the future, it's better to align the naming.